js/start.js

```diff
@@ -444,7 +457,40 @@ var loadProfile = function(landingRoute, onboarded) {
               app.simpleMessageModal.remove = () => {
                 throw new Error('This instance of the simpleMessageModal is globally shared ' +
                   'and should not be removed. When you are done with it, please close it.');
-              };              
+              };
+
+              // any lang changes after our app has loaded will need an app re-start to fully take effect
+              user.on('change:language', function(md, lang) {
+                var restartWarning;
+
+                restartWarning = new Dialog({
+                  title: window.polyglot.t('langChangeRestartTitle'),
+                  message: window.polyglot.t('langChangeRestartMessage'),
+                  buttons: [{
+                    text: 'Restart Later',
+                    fragment: 'restart-later'
+                  }, {
+                    text: 'Restart Now',
+                    fragment: 'restart-now'
+                  }]
+                }).on('click-restart-later', () => {
+                  restartWarning.close();
+                }).on('click-restart-now', () => {
+                  location.reload();
+                }).render().open();
+              });
+
+              // If navigating to a page you're already on, we'll hard refresh so
+              // a cached page is not served, which avoids the issue of it looking
+              // like nothing happened.
+              $(document).on('click', '[href^="#"]', (e) => {
```

Comment:
Would it be better to put this in the click handler at line 223, so there aren't two separate checks being done on all links?

What if going to the same page is triggered in JS instead of a click on a link?